x-pack/plugins/security_solution/server/endpoint/routes/metadata/handlers.ts

```diff
@@ -268,13 +268,9 @@ async function findAgent(
       throw new Error('esClient not found');
     }
 
-    const esClient = (metadataRequestContext?.esClient ??
-      metadataRequestContext.requestHandlerContext?.core.elasticsearch
-        .client) as IScopedClusterClient;
-
-    return await metadataRequestContext.endpointAppContextService
-      ?.getAgentService()
-      ?.getAgent(esClient.asCurrentUser, hostMetadata.elastic.agent.id);
+    return await metadataRequestContext.requestHandlerContext?.fleet?.agentService.asCurrentUser.getAgent(
+      hostMetadata.elastic.agent.id
+    );
```

Comment:
@paul-tavares Does this approach make sense to you before I continue updating the security solution code?

Reply:
yeah, it does. So we would no longer have the need to pass along.
I'm not the most familiar with `metadataRequestContext.requestHandlercontext`, but need to check how this new factory gets into the `EndpointAppContextService`. We use the services from that singleton in other areas of the server code (but maybe you are already looking into that?

Thanks for making these changes.

cc/ @pzl , @joeypoon , @ashokaditya  FYI ☝️

Reply:
> We use the services from that singleton in other areas of the server code (but maybe you are already looking into that?

Yes I'll update all the usages, just wanted an example to show before doing that work.